plugins/GSdx/GSSettingsDlg.cpp

```diff
@@ -712,7 +711,16 @@ void GSHacksDlg::OnInit()
 	CheckDlgButton(m_hWnd, IDC_FAST_TC_INV, theApp.GetConfigB("UserHacks_DisablePartialInvalidation"));
 	CheckDlgButton(m_hWnd, IDC_AUTO_FLUSH, theApp.GetConfigB("UserHacks_AutoFlush"));
 	CheckDlgButton(m_hWnd, IDC_UNSCALE_POINT_LINE, theApp.GetConfigB("UserHacks_unscale_point_line"));
-
+	if(ogl && theApp.m_gs_offset_hack.size()==2)
+	{
+		theApp.m_gs_offset_hack.push_back(GSSetting(2,  "Special (Texture)", ""));
+		theApp.m_gs_offset_hack.push_back(GSSetting(3,  "Special (Texture - aggressive)", ""));
+	}
+	else if (!ogl && theApp.m_gs_offset_hack.size()==4)
+	{
+		theApp.m_gs_offset_hack.erase(theApp.m_gs_offset_hack.begin() + 2,theApp.m_gs_offset_hack.begin() + 4); //We need to remove element 4 because function deletes elements between first and second including first but excluding second.
+	}
```

Comment:
I don't like the checks for vector size here (``theApp.m_gs_offset_hack.size()==4`` & ``theApp.m_gs_offset_hack.size()==2`` ). There are other alternatives to do this without checking for the vector size, I'd prefer such implementation.